ws-server/src/server.rs

```diff
@@ -598,6 +610,11 @@ impl Builder {
 		self
 	}
 
+	/// TODO: (dp)
+	pub fn with_middleware(mut self, m: M) -> Self {
+		self.middleware = m;
+		self
+	}
```

Comment:
```suggestion
	pub fn with_middleware<T>(self, middleware: T) -> Builder<T> {
		Builder {
			settings: self.settings,
			resources: self.resources,
			middleware,
		}
	}
```

Reply:
Ty!

I wonder if we should move away from `Builder::default().set_something().build()` and instead have `Builder::with_middleware(MMM).set_something().build()`/``Builder::new().set_something().build()`. Wdyt? This "default" is increasingly awkward imo.